### ISSUE TITLE
Dreambooth  example script only uses CPU for training instead of GPU

### ISSUE THREAD

**Issue by Hendrixer** [huggingface/diffusers#2156]
Hey team, I'm trying to execute the example dreambooth script to train, and when it initializes I see that it only uses the CPU as the device, instead of my GPU, which is a 4090.  I checked the accelerate config and where I set `'use_cpu'`to `false`. I'm new to ML and Python in general, coming from a web / mobile eng background. Anything I'm missing here? Thanks

Here are my system specs:
OS: Windows 11
GPU: 4090 FE
CPU: 3800X 3D
Py: 3.9
Cuda: 11.8

Accelerate Config
```json
{
  "command_file": null,
  "commands": null,
  "compute_environment": "LOCAL_MACHINE",
  "deepspeed_config": {},
  "distributed_type": "NO",
  "downcast_bf16": false,
  "dynamo_backend": "NO",
  "fsdp_config": {},
  "gpu_ids": "0",
  "machine_rank": 0,
  "main_process_ip": null,
  "main_process_port": null,
  "main_training_function": "main",
  "megatron_lm_config": {},
  "mixed_precision": "no",
  "num_machines": 1,
  "num_processes": 1,
  "rdzv_backend": "static",
  "same_network": false,
  "tpu_name": null,
  "tpu_zone": null,
  "use_cpu": false
}
```

Training script
```bat
@echo off

set MODEL_NAME=runwayml/stable-diffusion-v1-5
set INSTANCE_DIR=".\brian\512"
set CLASS_DIR=".\classimages\man\512"
set OUTPUT_DIR=".\models"
set CUDA_VISIBLE_DEVICES=0
set NVIDIA_VISIBLE_DEVICES=0
accelerate launch ^
train_dreambooth.py ^
--pretrained_model_name_or_path=%MODEL_NAME% ^
--checkpointing_steps=300 ^
--train_text_encoder ^
--instance_data_dir=%INSTANCE_DIR% ^
--class_data_dir=%CLASS_DIR% ^
--output_dir=%OUTPUT_DIR% ^
--with_prior_preservation ^
--prior_loss_weight=1.0 ^
--instance_prompt="a photo of sks man" ^
--class_prompt="a photo of man" ^
--resolution=512 ^
--train_batch_size=2 ^
--gradient_accumulation_steps=1 ^
--learning_rate=2e-6 ^
--lr_scheduler=constant ^
--lr_warmup_steps=0 ^
--num_class_images=200 ^
--max_train_steps=1200
```

I'm guessing there are some args I need to pass to the script, or some more dependencies like xformers?

**Comment by ameeratgithub** [comment]
I'm facing the same issue. Did you find any solution?

**Comment by Hendrixer** [comment]
@ameeratgithub I did. Turns out, even using the Pytorch web ui to download the correct version, it would still download the CPU only version of Pytorch. I was on cuda 11.8. I since went back to 11.6 and downloaded the correct Pytorch versions. I can now get the script to run using my 4090, however, as soon as it starts, I get an out of memory error :(

**Comment by Hendrixer** [comment]
> I'm facing the same issue. Did you find any solution?

@ameeratgithub I did. Turns out, even using the Pytorch web ui to download the correct version, it would still download the CPU only version of Pytorch. I was on cuda 11.8. I since went back to 11.6 and downloaded the correct Pytorch versions. I can now get the script to run using my 4090, however, as soon as it starts, I get an out of memory error :(
